app/bingey-ui/src/components/Watchlist/Watchlist.jsx

```diff
@@ -58,7 +58,7 @@ export const Watchlist = () => {
       setWatchlist(await api.getWatchlistById(id));
     };
     loadWatchlist();
-  }, [watchlist]);
+  }, [id]);
```

Comment:
Dependency on `watchlist` was causing endless rerendering as it was also changing it inside